golem-shard-manager/src/model.rs

```diff
@@ -79,11 +78,30 @@ impl Pod {
             ip: source_ip,
         };
 
-        let resolved = pod.address()?.map(|sa| sa.ip()).collect::<Vec<_>>();
-        if !resolved.contains(&source_ip) {
-            warn!("Host mismatch between registration message and resolved message source. Provided: {resolved:?}; source: {source_ip:?}");
+        match pod.address() {
+            Ok(resolved_addresses) => {
+                let resolved_addresses = resolved_addresses.map(|sa| sa.ip()).collect::<Vec<_>>();
+
+                if !resolved_addresses.contains(&source_ip) {
+                    warn!(
+                        pod= %pod,
+                        resolved_adresses = ?resolved_addresses,
+                        source_ip = %source_ip,
+                        "Host mismatch between registration message and resolved message source"
+                    );
+                }
+
+                Ok(pod)
+            }
+            Err(error) => {
+                error!(
+                    pod=?pod,
```

Comment:
here should be possible to use Display instead of Debug for arguments

Reply:
https://github.com/golemcloud/golem/pull/690/commits/0a86347c73d43333b0b3c22abe0447b3fe41bebe